### PR TITLE
ndmp: introduce workaround for isilon 9.1.0.0 'Invalid nlist.tape_offset -1' error

### PR DESCRIPTION
**Pull request by pstorz**
(Backport of #1043)
The isilon 9.1.0.0 release seems to have added a sanity check in ndmp
that checks that the tape offsets are on 512B boundaries.

The following error is issued when a restore is triggered to a isilon
9.1.0.0 system:

  'Invalid nlist.tape_offset -1 at index 1 - tape offset not aligned at 512B boundary'

However, -1 is an allowed value (NDMP_INVALID_U_QUAD) and was accepted
on every NDMP implementation we tested against including older isilon
versions.

To fix the problem, the isilon needs to check for NDMP_INVALID_U_QUAD
before it checks for the 512B boundary to fix the problem.

Otherwise we could enable this workaround depending on a new
configuration option.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [x] Decision taken that a system- or unittest is required (if not, then remove this paragraph)
- [x] The decision towards a systemtest is reasonable compared to a unittest
- [x] Testname matches exactly what is being tested
- [x] Output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: core/src/dird/dird_conf.cc [hunk 1]
```diff
@@ -149,6 +149,7 @@ static ResourceItem dir_items[] = {
   { "KeyEncryptionKey", CFG_TYPE_AUTOPASSWORD, ITEM(res_dir, keyencrkey), 1, 0, NULL, NULL, NULL },
   { "NdmpSnooping", CFG_TYPE_BOOL, ITEM(res_dir, ndmp_snooping), 0, 0, NULL, "13.2.0-", NULL },
   { "NdmpLogLevel", CFG_TYPE_PINT32, ITEM(res_dir, ndmp_loglevel), 0, CFG_ITEM_DEFAULT, "4", "13.2.0-", NULL },
+  { "NdmpNamelistFhinfoSetZeroForInvalidUquad", CFG_TYPE_BOOL, ITEM(res_dir, ndmp_fhinfo_set_zero_for_invalid_u_quad), 0, 0, NULL, "21.0.1-", NULL },
```
Comment (bruno-at-bareos) [inline core/src/dird/dird_conf.cc:152]:
```suggestion
  { "NdmpNamelistFhinfoSetZeroForInvalidUquad", CFG_TYPE_BOOL, ITEM(res_dir, ndmp_fhinfo_set_zero_for_invalid_u_quad), 0, CFG_ITEM_DEFAULT, "false", "21.0.1-", NULL },
```

**Review by bruno-at-bareos — CHANGES_REQUESTED** [review]
I've made the suggestion for having the default value for the documentation.

File: docs/manuals/source/include/autogenerated/bareos-dir-config-schema.json [hunk 2]
```diff
@@ -188,6 +188,12 @@
           "equals": true,
           "versions": "13.2.0-"
         },
+        "NdmpNamelistFhinfoSetZeroForInvalidUquad": {
+          "datatype": "BOOLEAN",
+          "code": 0,
+          "equals": true,
+          "versions": "21.0.1-"
```
Comment (bruno-at-bareos) [inline docs/manuals/source/include/autogenerated/bareos-dir-config-schema.json:195]:
```suggestion
          "versions": "20.0.6-"
```

**Review by bruno-at-bareos — DISMISSED** [review]
Please check the proposal suggestions, (version change to 20.0.6 due to backport)
Also add a CHANGELOG entry.

**Review by arogge — APPROVED** [review]

**Pull request merged**